pinc/gradual.inc

```diff
@@ -227,3 +228,144 @@ function maybe_output_new_proofer_project_message($project)
         echo "</div>";
     }
 }
+
+/*
+ * Output a callout and encourage the user to work in the round with the
+ * highest backlog that they can work in.
+ */
+function encourage_highest_round($username, $round_id = null)
+{
+    global $code_url, $Round_for_round_id_, $ELR_round, $ACCESS_CRITERIA;
+
+    // This URL fragment gets used a lot in this file
+    $round_url = "$code_url/tools/proofers/round.php?round_id=";
+
+    // get the backlog stats for all rounds sorted largest first
+    $backlog_stats = get_round_backlog_stats(array_keys($Round_for_round_id_));
+    arsort($backlog_stats);
+
+    // get all the rounds a user can work
+    $rounds_can_work_in = array_keys(get_stages_user_can_work_in($username));
+
+    // loop through the needs, highest to lowest, and point them to the
+    // one where they can make the most impact with their current access
+    foreach (array_keys($backlog_stats) as $round_target) {
```

Comment:
Just curious about the impact of the db call. Is this a popular page with this new db call that may impact other scenarios performance? I assume not, just a thought.

Reply:
Fair question and I've thought about it. This is the same query that is used to generate the round backlog graphs that are in news items on the Activity Hub and the P1 round page and so it seems to be pretty quick. I'm already eyeing generalizing the `query_graph_cache()` and making use of it here and on the My Recommendations page, but I don't want to optimize too soon.